src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -26,31 +26,52 @@ public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
 
         internal static bool IsEnabled()
         {
-            //check if someone listens to HttpHandlerDiagnosticListener
-            return s_diagnosticListener.IsEnabled();
+            // check if there is a parent Activity (and propagation is not suppressed)
+            // or if someone listens to HttpHandlerDiagnosticListener
+            return EnableActivityPropagation && (Activity.Current != null || s_diagnosticListener.IsEnabled());
         }
 
         protected internal override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
             CancellationToken cancellationToken)
         {
-            //HttpClientHandler is responsible to call DiagnosticsHandler.IsEnabled() before forwarding request here.
-            //This code will not be reached if no one listens to 'HttpHandlerDiagnosticListener', unless consumer unsubscribes
-            //from DiagnosticListener right after the check. So some requests happening right after subscription starts
-            //might not be instrumented. Similarly, when consumer unsubscribes, extra requests might be instrumented
+            // HttpClientHandler is responsible to call static DiagnosticsHandler.IsEnabled() before forwarding request here.
+            // This code will not be reached if Activity propagation is disabled unless consumer unsubscribes
+            // from DiagnosticListener right after the check. So some requests happening right after subscription starts
+            // might not be instrumented. Similarly, when consumer unsubscribes, extra requests might be instrumented
```

Comment:
Is this comment still accurate? It likes like will reach this even if s_diagnosticListener.IsEnabled is false.